backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -166,7 +182,28 @@ func (c *ClientManager) Authenticators() []auth.Authenticator {
 	return c.authenticators
 }
 
+func (c *ClientManager) ControllerClient() ctrlclient.Client {
+	return c.controllerClient
+}
+
 func (c *ClientManager) init() {
+	glog.Info("Initializing controller client...")
+	restConfig, err := util.GetKubernetesConfig()
+	if err != nil {
+		panic(fmt.Sprintf("Failed to initialize the RestConfig: %v", err))
```

Comment:
I think it'd be better to return errors here rather than panic. Then let the calling function panic if it can't continue.

Since this is a private method, changing the signature to return an error is no problem.